lib/syskit/network_generation/logger.rb

```diff
@@ -224,18 +224,9 @@ def self.add_logging_to_network(engine, work_plan)
 
                 # Finally, select 'default' as configuration for all
                 # remaining tasks that do not have a 'conf' argument set
-                work_plan.find_local_tasks(logger_model)
-                         .each do |task|
-                    unless task.arguments[:conf]
-                        task.arguments[:conf] = ["default"]
-                    end
-                end
-
-                # Mark as permanent any currently running logger
-                work_plan.find_tasks(logger_model)
-                         .not_finished
-                         .to_a.each do |t|
-                    work_plan.add_permanent_task(t)
+                required_loggers.each do |task|
+                    task.arguments[:conf] = ["default"] unless task.arguments[:conf]
+                    work_plan.add_permanent_task(task)
```

Comment:
Don't you have to check if "task" is finished anymore?

Reply:
Since I'm iterating over required_loggers, I am essentially assuming that we aren't relying on stopped loggers.

Which might or might not be the case... Not sure there is a test.